### ISSUE TITLE
Add Technical Documentation on Terraform Recipes

### ISSUE THREAD

**Issue by kachawla** [radius-project/radius#6825]
Based on some of the user questions I have seen, it would be beneficial to include high-level architecture documentation for Terraform Recipes, particularly focusing on areas like state file management. Design reviews for Recipes and Terraform were completed before moving the design docs to the GitHub repo. So it would be helpful to add documentation for both of these to the repo.

[AB#10451](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10451)

**Comment by shalabhms** [comment]
triage: @AaronCrawfis , is the issue important to be addressed in as part of  doc?

**Comment by AaronCrawfis** [comment]
> triage: @AaronCrawfis , is the issue important to be addressed in as part of doc?

yeah a concept doc in the Radius docs that talks about how Terraform deployments work would be helpful. Could live next to our architecture docs. Or a backdated design review doc that we could reference might be just as good.
